src/base/bittorrent/session.cpp

```diff
@@ -1949,13 +1916,11 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
         }
     }
 
-    // Remove it from torrent resume directory
-    QDir resumeDataDir(m_resumeFolderPath);
-    QStringList filters;
-    filters << QString("%1.*").arg(torrent->hash());
-    const QStringList files = resumeDataDir.entryList(filters, QDir::Files, QDir::Unsorted);
-    for (const QString &file : files)
-        Utils::Fs::forceRemove(resumeDataDir.absoluteFilePath(file));
+    // Remove it from the db
+    using namespace DB;
+    QSqlDatabase db = QSqlDatabase::database(MAIN_CONNECTION_NAME);
+    QSqlQuery query(db);
+    query.exec(QString("DELETE FROM %1 WHERE %2 = '%3'").arg(TABLE_NAME, COL_HASH, torrent->hash()));
```

Comment:
are you constantly executing these queries?  if so, it's good to cache them, and then update the args using hte cache query in the future.  it saves expensive costs in sqlite parsing and evaluating these strings. (note: only important if this is done a lot).

Reply:
I was talking in a general sense. Not about this query in particular.  For example, if you are constantly quering the DB, it can help significantly to cache the query.  

For example, say a user has many thousand torrents and you're going to execute the same query back to back for each torrent, then this can help quite a bit.

I agree it doesn't need to be done for something that happens rarely

Reply:
>I was talking in a general sense. Not about this query in particular.

Sorry. I just saw this suggestion having a general meaning in your other comment. Here it was attached to a specific place of the code, so it looks like it relates to this particular case.

Having prepared statement is a good idea in case of frequently invoked query.